.github/workflows/build.yaml

```diff
@@ -0,0 +1,36 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  
+jobs:
+  test-fedora:
+    name: fedora
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Setup Podman
+        run: |
+          sudo apt update
+          sudo apt-get -y install podman
+          podman pull fedora:latest
+      - name: Get source
+        uses: actions/checkout@v3
+        with:
+          path: 'x86-simd-sort'
+      - name: Create container and build
+        run: |
+          {
+              echo 'FROM fedora:latest'
+              echo 'RUN dnf -y update'
+              echo 'RUN dnf -y install gcc-c++ gtest-devel meson'
```

Comment:
Install g++-12 so that all the tests and benchmarks get built.

Reply:
Next release of Fedora (38) uses GCC 13, it seems better not to fix a particular version, but perhaps test for a range of versions.